azurerm/internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -0,0 +1,344 @@
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(machinelearningservices.SystemAssigned),
+							}, false),
+						},
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"scale_settings": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_node_count": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"min_node_count": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"node_idle_time_before_scale_down": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"subnet_resource_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"tags": tags.ForceNewSchema(),
+		},
+	}
+}
+
+func resourceComputeClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	mlComputeClient := meta.(*clients.Client).MachineLearning.MachineLearningComputeClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+
+	// Get Machine Learning Workspace Name and Resource Group from ID
+	workspaceID, err := parse.WorkspaceID(d.Get("machine_learning_workspace_id").(string))
+	if err != nil {
+		return err
+	}
+
+	existing, err := mlComputeClient.Get(ctx, workspaceID.ResourceGroup, workspaceID.Name, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("error checking for existing Compute Cluster %q in Workspace %q (Resource Group %q): %s",
+				name, workspaceID.Name, workspaceID.ResourceGroup, err)
+		}
+	}
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_machine_learning_compute_cluster", *existing.ID)
+	}
+
+	computeClusterProperties := machinelearningservices.AmlCompute{
+		Properties: &machinelearningservices.AmlComputeProperties{
+			VMSize:        utils.String(d.Get("vm_size").(string)),
+			VMPriority:    machinelearningservices.VMPriority(d.Get("vm_priority").(string)),
+			ScaleSettings: expandScaleSettings(d.Get("scale_settings").([]interface{})),
+			Subnet:        &machinelearningservices.ResourceID{ID: utils.String(d.Get("subnet_resource_id").(string))},
+		},
+		ComputeLocation: utils.String(d.Get("location").(string)),
+		Description:     utils.String(d.Get("description").(string)),
+		ComputeType:     "ComputeTypeAmlCompute1",
```

Comment:
This should be `machinelearningservices.ComputeTypeAmlCompute1` instead of hard-code this string here. 
However, the value of discriminator can be omitted, which could be implied by the SDK from the name if its type.
Please see the marshaller of this type for more information: https://github.com/Azure/azure-sdk-for-go/blob/18d0067e8edb39adc029ec410c1f468ac2fc5fa2/services/machinelearningservices/mgmt/2020-04-01/machinelearningservices/models.go#L250
in this marshaller, any value you assigned to `ComputeType` will be overridden

Reply:
done 👍